rajawali/src/main/java/org/rajawali3d/materials/shaders/fragments/specular/PhongFragmentShaderFragment.java

```diff
@@ -43,16 +43,13 @@
 	
 	private List<ALight> mLights;
 	
-	public PhongFragmentShaderFragment(List<ALight> lights, int specularColor, float shininess) {
+	public PhongFragmentShaderFragment(List<ALight> lights, float[] specularColor, float shininess) {
 		this(lights, specularColor, shininess, 1, null);
 	}
 	
-	public PhongFragmentShaderFragment(List<ALight> lights, int specularColor, float shininess, float specularIntensity, List<ATexture> textures) {
+	public PhongFragmentShaderFragment(List<ALight> lights, float[] specularColor, float shininess, float specularIntensity, List<ATexture> textures) {
 		super(textures);
-		mSpecularColor = new float[] { 1, 1, 1 };
-		mSpecularColor[0] = (float)Color.red(specularColor) / 255.f;
-		mSpecularColor[1] = (float)Color.green(specularColor) / 255.f;
-		mSpecularColor[2] = (float)Color.blue(specularColor) / 255.f;
+		mSpecularColor = mSpecularColor;
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self assignment of value (SA_SELF_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-assignment-of-field-sa-field-self-assignment).
This method contains a self assignment of a field; e.g.
Such assignments are useless, and may indicate a logic error or typo.